src/irmin-pack/pack_dict.ml

```diff
@@ -20,4 +20,6 @@ let (`Staged v) =
     "store.dict"
 
 let v ?fresh ?readonly ?(capacity = 100_000) root =
-  v capacity ?fresh ?readonly root
+  let dict = v capacity ?fresh ?readonly root in
+  if readonly = Some true then ro_sync dict;
```

Comment:
What's the purpose of syncing immediately after creating the instance? Is this to do with shared instances? (if so, does that present problems for the "RO instances should not have data changed under their feet" goal?)

Reply:
it's not needed, indeed (removed it).